tests/unit/test_utils.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2020 Ubuntu
+# See LICENSE file for licensing details.
+
+import os
+import pathlib
+import tempfile
+import unittest
+from uuid import uuid4
+
+import utils
+
+
+class TestUtils(unittest.TestCase):
+    def test_find_packages_by_indices(self):
+        nums = 3
+        expected_outputs = set()
+        returned_output = set()
+        # Generate a test file structure for testing
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            tmpdirname = pathlib.Path(tmpdirname)
+            for i in range(nums):
+                any_dirs = str(i)
+                dists = tmpdirname / any_dirs / "dists"
+                dists.mkdir(parents=True)
+                pool = tmpdirname / any_dirs / "pool"
+                pool.mkdir(parents=True)
+                archive_root = pool.parent.absolute()
+                distros = [dists / str(i) for i in range(nums)]
+                for distro in distros:
+                    distro.mkdir(parents=True)
+                    index = distro / "Packages"
+                    with index.open(mode="w") as f:
+                        name = "{}.deb".format(str(uuid4))
+                        f.write("Filename: {}\n".format(name))
+                        expected_outputs.add(archive_root / name)
+
+            # get returned output
+            for archive_root, indices in utils.locate_package_indices(tmpdirname):
+                returned_output |= utils.find_packages_by_indices(
+                    indices, base=archive_root
+                )
+
+        self.assertEqual(sorted(expected_outputs), sorted(returned_output))
+
+    def test_locate_package_indices(self):
+        nums = 3
+        expected_outputs = []
+        # Generate a test file structure for testing
+        with tempfile.TemporaryDirectory() as tmpdirname:
```

Comment:
The code to generate temporary directory seems duplicated. We can try to put the testing envirnoment with the unit test together.

Reply:
same above